core/src/subgraph/runner.rs

```diff
@@ -1361,7 +1349,7 @@ where
             .deployment_head
             .set(subgraph_ptr.number as f64);
 
-        self.revert_state(subgraph_ptr.number)?;
+        self.revert_state_to(revert_to_ptr.number)?;
```

Comment:
This is where reverting by more than one block came into play, right?

Reply:
Yes this _and_ the implementation in `dynds/private.rs` were both wrong I think.